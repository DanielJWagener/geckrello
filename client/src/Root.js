// Simply creates a Redux store and passes it in to a provider tag.
// This logic was originally in index.js, but we've movied it here to
// make our testing process easier.
import React from "react";
import { Provider } from "react-redux";
import { createStore, applyMiddleware, compose } from "redux";
import reduxThunk from "redux-thunk";
import reducers from "./reducers";

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

const store = createStore(
  reducers,
  {},
  composeEnhancers(applyMiddleware(reduxThunk))
);

export default ({ children }) => {
  return <Provider store={store}>{children}</Provider>;
};
