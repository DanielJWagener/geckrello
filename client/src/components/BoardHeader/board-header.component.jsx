import React, { Component } from "react";

import BoardTitle from "../BoardTitle/board-title.component";
import MenuToggler from "../MenuToggler/menu-toggler.component";
import BoardMenuWrapper from "../BoardMenuWrapper/board-menu-wrapper.component";

import "./board-header.styles.scss";

export class BoardHeader extends Component {
  state = {
    menuHidden: true,
  };

  toggleMenu = () => {
    if (this.state.menuHidden) {
      this.setState({ menuHidden: false });
    } else {
      this.setState({ menuHidden: true });
    }
  };

  render() {
    return (
      <div className="board-header">
        <BoardTitle board={this.props.board} />
        <MenuToggler
          toggleMenu={this.toggleMenu}
          menuHidden={this.state.menuHidden}
        />

        <BoardMenuWrapper
          menuHidden={this.state.menuHidden}
          toggleMenu={this.toggleMenu}
        />
      </div>
    );
  }
}

export default BoardHeader;
