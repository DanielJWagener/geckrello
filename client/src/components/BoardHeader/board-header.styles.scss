@import "../../sass/abstracts/_variables.scss";

.board-header {
  display: flex;
  justify-content: space-between;
  position: relative;
  padding: 0.5rem 0;
  background-color: var(--color-primary);
  height: $board-header-height;

  &--left {
    display: flex;
    align-items: center;
  }
}
