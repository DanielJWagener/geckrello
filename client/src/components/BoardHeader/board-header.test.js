import { shallow } from "enzyme";
import React from "react";
import BoardHeader from "./board-header.component";
import Root from "../../Root";

it("renders BoardHeader component", () => {
  expect(
    shallow(
      <Root>
        <BoardHeader />
      </Root>
    )
  ).toMatchSnapshot();
});
