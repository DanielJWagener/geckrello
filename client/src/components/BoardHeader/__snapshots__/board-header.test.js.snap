// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders BoardHeader component 1`] = `
"<div className=\\"board-header\\">
  <Connect(BoardTitle) board={[undefined]} />
  <MenuToggler toggleMenu={[Function]} menuHidden={true} />
  <BoardMenuWrapper menuHidden={true} toggleMenu={[Function]} />
</div>"
`;
